### ISSUE TITLE
Experimental feature failure - Unknown 'ValidateResource' shadow value for "aws_elb"

### ISSUE THREAD

**Issue by tomwilkie** [hashicorp/terraform#10440]
I deleted some ec2_instances, which in turn triggered an update of the ELB, and got the following error:

```
2 error(s) occurred:

* apply operation: Unknown 'ValidateResource' shadow value for "aws_elb": &errors.errorString{s:"shadow closed"}
* apply operation: Diff "module.infra.aws_elb.main" had unequal states (real, then shadow):

&terraform.ResourceConfig{ComputedKeys:[]string(nil), Raw:map[string]interface {}{"health_check":[]map[string]interface {}{map[string]interface {}{"healthy_threshold":2, "unhealthy_threshold":6, "timeout":5, "interval":10, "target":"TCP:30080"}}, "instances":[]interface {}{"${aws_instance.minion.*.id}"}, "name":"${var.elb_name}", "subnets":[]interface {}{"${aws_subnet.main.*.id}"}, "cross_zone_load_balancing":"false", "tags":[]map[string]interface {}{map[string]interface {}{"kubernetes.io/service-name":"default/frontend", "KubernetesCluster":"${var.cluster_name}"}}, "listener":[]map[string]interface {}{map[string]interface {}{"instance_protocol":"tcp", "lb_port":80, "lb_protocol":"tcp", "instance_port":30080}, map[string]interface {}{"instance_port":30443, "instance_protocol":"tcp", "lb_port":443, "lb_protocol":"tcp"}}}, Config:map[string]interface {}{"cross_zone_load_balancing":"false", "tags":[]map[string]interface {}{map[string]interface {}{"KubernetesCluster":"kubernetes_dev_2", "kubernetes.io/service-name":"default/frontend"}}, "listener":[]map[string]interface {}{map[string]interface {}{"instance_port":30080, "instance_protocol":"tcp", "lb_port":80, "lb_protocol":"tcp"}, map[string]interface {}{"lb_port":443, "lb_protocol":"tcp", "instance_port":30443, "instance_protocol":"tcp"}}, "health_check":[]map[string]interface {}{map[string]interface {}{"healthy_threshold":2, "unhealthy_threshold":6, "timeout":5, "interval":10, "target":"TCP:30080"}}, "instances":[]interface {}{"i-6c81385c", "i-ad08da35", "i-73d4dae0", "i-f2df5cc2"}, "name":"ab12448b348fa11e69d561228d43a32f", "subnets":[]interface {}{"subnet-b3c31c9e", "subnet-c0f78489", "subnet-bc1a9fe7"}}, raw:(*config.RawConfig)(0xc42039ede0)}

&terraform.ResourceConfig{ComputedKeys:[]string(nil), Raw:map[string]interface {}{"name":"${var.elb_name}", "subnets":[]interface {}{"${aws_subnet.main.*.id}"}, "cross_zone_load_balancing":"false", "tags":[]map[string]interface {}{map[string]interface {}{"kubernetes.io/service-name":"default/frontend", "KubernetesCluster":"${var.cluster_name}"}}, "listener":[]map[string]interface {}{map[string]interface {}{"instance_port":30080, "instance_protocol":"tcp", "lb_port":80, "lb_protocol":"tcp"}, map[string]interface {}{"lb_port":443, "lb_protocol":"tcp", "instance_port":30443, "instance_protocol":"tcp"}}, "health_check":[]map[string]interface {}{map[string]interface {}{"target":"TCP:30080", "healthy_threshold":2, "unhealthy_threshold":6, "timeout":5, "interval":10}}, "instances":[]interface {}{"${aws_instance.minion.*.id}"}}, Config:map[string]interface {}{"name":"ab12448b348fa11e69d561228d43a32f", "subnets":[]interface {}{"subnet-b3c31c9e", "subnet-c0f78489", "subnet-bc1a9fe7"}, "cross_zone_load_balancing":"false", "tags":[]map[string]interface {}{map[string]interface {}{"KubernetesCluster":"kubernetes_dev_2", "kubernetes.io/service-name":"default/frontend"}}, "listener":[]map[string]interface {}{map[string]interface {}{"lb_protocol":"tcp", "instance_port":30080, "instance_protocol":"tcp", "lb_port":80}, map[string]interface {}{"instance_protocol":"tcp", "lb_port":443, "lb_protocol":"tcp", "instance_port":30443}}, "health_check":[]map[string]interface {}{map[string]interface {}{"timeout":5, "interval":10, "target":"TCP:30080", "healthy_threshold":2, "unhealthy_threshold":6}}, "instances":[]interface {}{"i-6c81385c", "i-ad08da35", "i-73d4dae0", "i-f2df5cc2", "i-ae08da36", "i-e0d5db73", "i-e8de5dd8", "i-fc0bd964", "i-e3d5db70", "i-93b02a9d"}}, raw:(*config.RawConfig)(0xc42057aa20)}
```

The ELB is defined like this:

```
resource "aws_elb" "main" {
    name = "${var.elb_name}"
    subnets = ["${aws_subnet.main.*.id}"]
    instances = ["${aws_instance.minion.*.id}"]
    ...
}
```

**Comment by tomwilkie** [comment]
Probably related to #9967, but that ones gone cold, so I thought I'd open another.

Am using terraform 0.7.13:
```
$ terraform -v
Terraform v0.7.13
```

**Comment by mitchellh** [comment]
The exact diff is that the shadow state during diff has a number more EC2 instances (it isn't reflecting what I assume are the deleted instances). 

I'll try to reproduce this but it'd be very helpful if you can get some reproduction steps. Thanks!

**Comment by tomwilkie** [comment]
I'm just about to release the source for the terraform modules that made this, will let you know when.

**Comment by tomwilkie** [comment]
@mitchellh heres the module that produced this error: https://github.com/weaveworks/terraform-kubernetes

I expected the following commands to reproduce the error:
```
# start with num_minions = 2
$ terraform apply
...
# set num_minions = 1
$ terraform apply
...
```

But they didn't.  I then did this:
```
# start with num_minions = 2
$ terraform apply
...
# set num_minions = 1
$ terraform plan -out=foo
$ terraform apply foo
```

And it did.  Hopefully that helps?

```
2 error(s) occurred:

* apply operation: Unknown 'ValidateResource' shadow value for "aws_elb": &errors.errorString{s:"shadow closed"}
* apply operation: Diff "module.infra.aws_elb.main" had unequal states (real, then shadow):

&terraform.ResourceConfig{ComputedKeys:[]string(nil), Raw:map[string]interface {}{"tags":[]map[string]interface {}{map[string]interface {}{"KubernetesCluster":"${var.cluster_name}", "kubernetes.io/service-name":"default/frontend"}}, "listener":[]map[string]interface {}{map[string]interface {}{"instance_protocol":"tcp", "lb_port":80, "lb_protocol":"tcp", "instance_port":30080}, map[string]interface {}{"lb_protocol":"tcp", "instance_port":30443, "instance_protocol":"tcp", "lb_port":443}}, "health_check":[]map[string]interface {}{map[string]interface {}{"interval":10, "target":"TCP:30080", "healthy_threshold":2, "unhealthy_threshold":6, "timeout":5}}, "instances":[]interface {}{"${aws_instance.minion.*.id}"}, "name":"${var.elb_name}", "subnets":[]interface {}{"${aws_subnet.main.*.id}"}}, Config:map[string]interface {}{"instances":[]interface {}{"i-d70c3c3f"}, "name":"test", "subnets":[]interface {}{"subnet-fc5c0398"}, "tags":[]map[string]interface {}{map[string]interface {}{"kubernetes.io/service-name":"default/frontend", "KubernetesCluster":"toms_cluster"}}, "listener":[]map[string]interface {}{map[string]interface {}{"lb_port":80, "lb_protocol":"tcp", "instance_port":30080, "instance_protocol":"tcp"}, map[string]interface {}{"lb_protocol":"tcp", "instance_port":30443, "instance_protocol":"tcp", "lb_port":443}}, "health_check":[]map[string]interface {}{map[string]interface {}{"interval":10, "target":"TCP:30080", "healthy_threshold":2, "unhealthy_threshold":6, "timeout":5}}}, raw:(*config.RawConfig)(0xc420618ba0)}

&terraform.ResourceConfig{ComputedKeys:[]string(nil), Raw:map[string]interface {}{"instances":[]interface {}{"${aws_instance.minion.*.id}"}, "name":"${var.elb_name}", "subnets":[]interface {}{"${aws_subnet.main.*.id}"}, "tags":[]map[string]interface {}{map[string]interface {}{"KubernetesCluster":"${var.cluster_name}", "kubernetes.io/service-name":"default/frontend"}}, "listener":[]map[string]interface {}{map[string]interface {}{"instance_protocol":"tcp", "lb_port":80, "lb_protocol":"tcp", "instance_port":30080}, map[string]interface {}{"lb_port":443, "lb_protocol":"tcp", "instance_port":30443, "instance_protocol":"tcp"}}, "health_check":[]map[string]interface {}{map[string]interface {}{"target":"TCP:30080", "healthy_threshold":2, "unhealthy_threshold":6, "timeout":5, "interval":10}}}, Config:map[string]interface {}{"subnets":[]interface {}{"subnet-fc5c0398"}, "tags":[]map[string]interface {}{map[string]interface {}{"kubernetes.io/service-name":"default/frontend", "KubernetesCluster":"toms_cluster"}}, "listener":[]map[string]interface {}{map[string]interface {}{"lb_port":80, "lb_protocol":"tcp", "instance_port":30080, "instance_protocol":"tcp"}, map[string]interface {}{"lb_protocol":"tcp", "instance_port":30443, "instance_protocol":"tcp", "lb_port":443}}, "health_check":[]map[string]interface {}{map[string]interface {}{"healthy_threshold":2, "unhealthy_threshold":6, "timeout":5, "interval":10, "target":"TCP:30080"}}, "instances":[]interface {}{"i-d70c3c3f", "i-c00f3f28"}, "name":"test"}, raw:(*config.RawConfig)(0xc4203caf60)}
```

**Comment by mitchellh** [comment]
@tomwilkie Thanks for the great repro steps I think I've got it over here, debugging now.

**Comment by mitchellh** [comment]
Phew! Got a fix in in PR #10522. I've verified it with your reproduction.

This was a really important bug to find. Thanks for reporting it.

**Comment by tomwilkie** [comment]
Thanks you for fixing it @mitchellh!
